bin/mastarm

```diff
@@ -64,24 +64,49 @@ commander
   .option('--cloudfront', 'CloudFront Distribution ID to invalidate.')
   .option('--s3bucket', 'S3 Bucket to push to.')
   .action(function (entries, options) {
+    const commit = require('this-commit')()
+    const username = require('username')
+    const createLogger = require('../lib/logger')
+    const pkg = require('../lib/pkg')
     const pushToS3 = require('../lib/push-to-s3')
+
+    const url = pkg.repository.url.replace('.git', '')
+    const tag = `<${url}/commit/${commit}|${pkg.name}@${commit.slice(0, 6)}>:`
+    const user = username.sync()
     const config = loadConfig(process.cwd(), commander.config, commander.env)
     const get = util.makeGetFn([options, commander, config.settings])
+
+    const env = get('env')
+    const s3bucket = get('s3bucket')
+
     const files = util.parseEntries([...entries, ...(get('entries') || [])])
     util.assertEntriesExist(files)
-    Promise.all(files.map(pushToS3({
-      cloudfront: get('cloudfront'),
+
+    const log = createLogger({channel: config.env.SLACK_CHANNEL || '#devops', webhook: config.env.SLACK_WEBHOOK})
+    const minify = get('minify')
+    const cloudfront = get('cloudfront')
+    const push = pushToS3({
+      cloudfront,
       config,
-      env: get('env'),
-      minify: get('minify'),
-      s3bucket: get('s3bucket')
-    }))).then(() => {
-      console.log('Finished deploying!')
-      process.exit(0)
-    }).catch((err) => {
-      console.error('Error while deploying!')
-      console.error(err.stack)
-      process.exit(1)
+      env,
+      log,
+      minify,
+      s3bucket,
+      tag
+    })
+
+    log(
+`:construction: *${tag} deploy started by <@${user}>*
+:cloud: *cloudfront:* ${cloudfront}
+:hash: *commit:* ${commit}
+:seedling: *env:* ${env}
+:compression: *minify:* ${minify}
+:package: *s3bucket:* ${s3bucket}`
+    ).then(() => {
+      Promise
+        .all(files.map(push))
+        .then(() => log(`:rocket: ${tag} deploy finished!!!! :tada: :confetti_ball: :tada:`).then(() => process.exit(0)))
```

Comment:
These lines are really long, please break them up to be more readable in an editor.